src/mlpack/core/data/dictionary_encoding.hpp

```diff
@@ -0,0 +1,154 @@
+/**
+ * @file dictionary_encoding.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of dictionary encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DICTIONARY_ENCODING_HPP
+#define MLPACK_CORE_DATA_DICTIONARY_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <utility>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple Dictionary Enocding class
+ */
+class DictionaryEncoding
+{
+ public:
+  /**
+  * A function to create mapping from a given corpus.
+  * 
+  * @param input Corpus of text to encode.
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  *
+  */
+  template<typename TokenizerType>
+  void CreateMap(std::string& input, TokenizerType tokenizer);
+  /*
+  * Default Constructor
+  */
+  DictionaryEncoding() {}
+  /*
+  * Copy Constructor.
+  */
+  DictionaryEncoding(const DictionaryEncoding& oldObject);
+  /*
+  * Move Constructor.
+  */
+  DictionaryEncoding(DictionaryEncoding&& oldObject) = default;
+  /*
+  * Move Assignment Operator.
+  */  
+  DictionaryEncoding& operator= (DictionaryEncoding&& oldObject) = default;
+  /*
+  * Assignment Operator.
+  */
+  DictionaryEncoding operator= (const DictionaryEncoding& oldObject);
```

Comment:
I think the return value should be a reference (I mean `DictionaryEncoding&`).

Reply:
Yeah